test/uri/uri_recompose_test.cc

```diff
@@ -18,7 +18,7 @@ TEST(URI_recompose, urn) {
 
 TEST(URI_recompose, https_with_empty_fragment) {
   const sourcemeta::jsontoolkit::URI uri{"https://example.com/foo#"};
-  EXPECT_EQ(uri.recompose(), "https://example.com/foo#");
+  EXPECT_EQ(uri.recompose(), "https://example.com/foo");
```

Comment:
⚠️ @jviotti

Reply:
I think we should address these differences. In theory `https://example.com/foo#` and `https://example.com/foo` are different URIs. They may point to the same thing, but that should happen only when doing `canonicalize()`. Otherwise changing the URI that the user is giving us might be a recipe for disaster.